packages/sdk/src/contracts/prebuilt-implementations/marketplace.ts

```diff
@@ -296,6 +296,39 @@ export class Marketplace implements UpdateableNetwork {
     return this.contractWrapper.readContract.timeBuffer();
   }
 
+  /**
+   * Get all the offers
+   *
+   * @remarks Fetch all the offers for a specified direct listing.
+   * @example
+   * ```javascript
+   * const offers = await marketplaceContract.getOffers(listingId);
+   * const firstOffer = offers[0];
+   * ```
+   *
+   * @param listingId - the id of the listing to fetch offers for 
+   */
+   public async getOffers(
+    listingId: BigNumberish,
+  ): Promise<(Offer)[]> {
+    // get all new offer events from this contract
+    const events = await this.events.getEvents("NewOffer");
+    // get only the events for this listing id
+    const listingEvents = events.filter((e) => e.data.listingId.eq(listingId));
+    // derive the offers from the events
+    const offers = listingEvents.map((e): Offer  => {
+      return {
+        listingId: e.data.listingId,
+        buyerAddress: e.data.offeror,
+        quantityDesired: e.data.quantityWanted,
+        pricePerToken: (e.data.totalOfferAmount).div(e.data.quantityWanted),
+        currencyValue: e.data.totalOfferAmount,
```

Comment:
this type is wrong - currencyValue is an object.

check out what we do in `common/marketplace.ts` in the `mapOffer()` function

Reply:
I pulled in the mapOffer function and am now using that (see [cad4099](https://github.com/thirdweb-dev/js/pull/191/commits/cad40990589bacd37850e0cd7aa11c2f59ecc33d))